creusot-contracts/src/std/option.rs

```diff
@@ -210,6 +283,7 @@ extern_spec! {
                     }
                 }
 
+                // FIXME: cannot add a body, deref is mistranslated (in logic?)
                 #[ensures(match *self {
                     None => result == None,
                     Some(t) => result == Some(&t.deref()),
```

Comment:
huh this is weird... `Deref::deref` is not a logic function, but rather a program function. This should actually be rejected. Instead, we should add a `DerefLogic` trait, with the spec of the `Deref` function...

(There is an exception for `Box` and friends, but here we are considering the generic case.)

Reply:
Yes, I think the exception accepts too much (every instance of deref and deref_mut is accepted).
Actually, let me try to fix that...

Reply:
Ok, done: I also removed this function, since I think it was unusable in practice anyways. I definitely agree that we should have `DerefLogic` at some point though.